.NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeSetHandler.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Microsoft.Recognizers.Text.DataTypes.TimexExpression;
+using DateObject = System.DateTime;
+
+namespace Microsoft.Recognizers.Text.DateTime.Utilities
+{
+    public static class TasksModeSetHandler
+    {
+        public static DateTimeResolutionResult TasksModeResolveSet(ref DateTimeResolutionResult result, string innerTimex, DateTimeParseResult pr = null)
+        {
+            result.Timex = innerTimex;
+
+            result.FutureValue = result.PastValue = "Set: " + innerTimex;
+
+            if (pr != null)
+            {
+                DateTimeResolutionResult value = (DateTimeResolutionResult)pr.Value;
+                if (value.FutureValue != null)
+                {
+                    if (pr.TimexStr.EndsWith("WE"))
+                    {
+                        result.FutureValue = ((Tuple<DateObject, DateObject>)value.FutureValue).Item1;
+                        result.PastValue = ((Tuple<DateObject, DateObject>)value.PastValue).Item1;
+                    }
+                }
+            }
+
+            result.Success = true;
+
+            return result;
+        }
+
+        public static Dictionary<string, string> TasksModeGenerateResolutionSetParser(Dictionary<string, string> resolutionDic, string mod, string timex)
+        {
+            var res = new Dictionary<string, string>();
+
+            TasksModeAddAltSingleDateTimeToResolution(resolutionDic, TimeTypeConstants.DATETIMEALT, mod, res);
+            if (timex.StartsWith("P") && res.Count > 0)
```

Comment:
Please avoid hardcoding values and timex-related terms. Reuse constants or methods that manipulate expressions.

Reply:
added new variable in tasksmodeconstant file.